src/search/services/search.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@nestjs/common';
+import { Board } from 'src/boards/entities/board.entity';
+import { User } from 'src/users/entities/user.entity';
+import { EntityManager } from 'typeorm';
+
+@Injectable()
+export class SearchService {
+  constructor(private entityManager: EntityManager) {}
+  async searchBoardsAsHeadOrBody(searchQuery: string) {
+    const boardRepository = this.entityManager.getRepository(Board);
+
+    return boardRepository
+      .createQueryBuilder()
+      .select()
+      .where(`MATCH(head) AGAINST (:searchQuery)`, {
```

Comment:
where 문자열에 백틱으로 searchQuery string 으로 넣을수도있지만 이 패턴을 사용한게 굉장히 좋네요 만약
```suggestion
      .where(`MATCH(head) AGAINST (${searchQuery})`, {
```
처럼 된다면 sql injection 공격에 취약해지기때문입니다

Reply:
제 의도를 정확히 보셨네요